staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -65,8 +69,9 @@ var _ MounterForceUnmounter = &Mounter{}
 // mounterPath allows using an alternative to `/bin/mount` for mounting.
 func New(mounterPath string) Interface {
 	return &Mounter{
-		mounterPath: mounterPath,
-		withSystemd: detectSystemd(),
+		mounterPath:                mounterPath,
+		withSystemd:                detectSystemd(),
+		withSafeNotMountedBehavior: detectSafeNotMountedBehavior(),
```

Comment:
each time a new mounter is created, this function detectSafeNotMountedBehavior will be called, right? Is there any performance concern?

Reply:
My understanding is that the mounter *is* only created once at kubelet boot, here: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L366

I don't think there's a performance concern unless a mounter is being created frequently, like once a second. I'm not aware of any code paths that do such a thing. I don't think the performance characteristics of `detectSafeNotMountedBehavior` differ significantly from the existing `detectSystemd`.

Reply:
got it, then it should be good.

Reply:
btw: do you know how linux "umount" command works to check whether it is a mount point. It should also check mount table, right? Is it much more efficient?

Reply:
It uses `libmount`, which has heuristics to avoid a mtab check whenever possible. It's implementation of a mtab check is also more efficient than what's happening in this library. Relevant function: https://github.com/util-linux/util-linux/blob/master/libmount/src/context_umount.c#L47

Another aspect of this PR is removing redundant checks. Even if the mount point check in this library had the same performance as `umount`'s, part of the issue is that the check is happening twice.